### ISSUE TITLE
[Oculus] The app is closed after resuming the app from immersive mode and the crash dialog box is displayed after it's reopened

### ISSUE THREAD

**Issue by Softvision-RemusDranca** [MozillaReality/FirefoxReality#2017]
**[Notes]:**
- This issue is intermittently reproducible on the first minutes of using the app, but after using the app for ~ one hour the issue is continuously reproducible.

**[Affected Versions]:**
- Firefox Reality version 6-32891835 (f856c035)

**[Affected Platforms]:**
- Oculus Quest - Android 7.1.1
- Oculus Go - Android 7.1.2

**[Prerequisites]:**
- The headset is turned on and the browser is opened.

**[Steps to reproduce]:**
1. Open and Play a 360 video in immersive mode.
2. Click the "Oculus" button from the controller.
3. Click the "Resume" button from the prompt displayed and observe the behavior.

**[Expected results]:**
- The app is resumed.

**[Actual results]:**
- The app is closed and after it's manually reopen the crash dialog box is displayed.

**[Additional Notes]:**
- Attached are a screen recording along with the logcat and the crash report of the issue: 
	- Screen recording: [link](https://drive.google.com/open?id=1x9v6Qr0Dsqpx5mtjOOAj2gX-UHFkTl_Z)
	- Logcat: [link](https://drive.google.com/open?id=1xeXkvWlbSGiuIOMvWweuBOppwrvaEftI)
	- Crash report: [link](https://crash-stats.mozilla.com/report/index/ca44330c-fb4b-4397-af4a-50cbe0191017)

**Comment by bluemarvin** [comment]
https://tools.taskcluster.net/groups/NHmoIulEQQaf0W8PkuZbsw/tasks/NHmoIulEQQaf0W8PkuZbsw/runs/0/artifacts
These builds contain the software rendering change. Would you retest using this build to see if it resolves the issue? Thanks.

**Comment by bluemarvin** [comment]
@Softvision-RemusDranca have you had a chance to test the above APKs? We would like to get this into master if it resolves the issue.

**Comment by Softvision-RemusDranca** [comment]
I've verified this issue with the Firefox Reality 6 RC1 (a83b2bf9) build and it's no longer reproducible. Now, the app is resumed in immersive mode. Tested on Oculus Quest and Oculus Go devices.
Considering this I will close this issue as verified-fixed. 
Sorry for the late response on this.

**Comment by bluemarvin** [comment]
The updated APKs can be found here: https://community-tc.services.mozilla.com/tasks/UIZHWWEAQvWSK7Fs4MrzHA
You will need to click `Artifacts` to expand the list to access the APKs.

**Comment by Softvision-RemusDranca** [comment]
I've retested this issue using the build provided in the above comment, on Oculus Quest Here are the results:
- The crash dialog box was not displayed in none of the attempts while resuming the app.
- The app was always resumed in immersive mode.
- The main issues are when the app is quited from immersive mode and reopened. The app has very bad performance issues as follow:
	- The websites are hardly loading.
	- In 2 attempts the device was restarted. Here are 2 logcats recorded: [logcat 1](https://drive.google.com/open?id=1NYAlAu7_w3VxZoidrm1GBwRZt2MWnzZm); [logcat 2](https://drive.google.com/open?id=1NmloqJu1LArLXH-pFMa7kVXskTMzHNkg)
	- In 1 attempt the "Firefox Reality has timed up" prompt was displayed.
	- The app automatically closed after reopening it and trying to navigate. Here is a video recording: [link](https://drive.google.com/open?id=1t3L4vUdgVweYNmje-NXxMOJ4OVdb8swS)

**Comment by MortimerGoro** [comment]
I think the problems after the app is quited are related to the restore issue, which was fixed in: https://github.com/MozillaReality/FirefoxReality/pull/2181

I can't locally reproduce the perf issue (but I'm testing with the fix included)

**Comment by bluemarvin** [comment]
We've decided to hold off on landing the software rendering due to the high risk of changing at this point. If we see a lot of crash reports from v6 for this issue we can look at landing for v7 so we have more time to test it.

**Comment by heres-maria** [comment]
I have verified this issue and it's no longer reproducible with the Firefox Reality version 7 RC3 (7b66fb14) build. Neither the Oculus Go, nor the Oculus Guest experienced any crashes, restarts or prompts and there was no noticeable difference in load time for websites afterwards for either resuming or quitting and reopening the app.
Tested on the Oculus Quest (Android 7.1.1) and Oculus Go (Android 7.1.2).
